[ETL-602] Compile documentation for recover (#105)
Sage-Bionetworks/recover: CONTRIBUTING.md

### Post review

> Once the pull request has been approved, we expect _you_ to merge. Although this pulls your commits into the `main` branch, it does not yet deploy your changes to the `main` production pipeline. RECOVER data has FISMA restrictions, but only our production account is FISMA compliant. Since there is no guarantee that the test data provided to us (which doesn't have FISMA restrictions) perfectly models the production dataset, we maintain a `staging` namespace in the production account which enables us to test changes on production data before pulling those changes into the `main` namespace. Merging into `main` will deploy the changes to the `staging` namespace. To complete deployment to the `main` namespace of production, we push a new tag with a specific format to this repository, which will trigger [this GitHub action](.github/workflows/upload-and-deploy-to-prod-main.yaml). There is a diagram of this process [here](https://sagebionetworks.jira.com/wiki/spaces/IBC/pages/2863202319/ETL-390+RECOVER+Integration+Testing#Implementation). See [release process](#release-process) for more details on this.

Reviewer: Could you add something here that mentions `Squash and merge`?

https://sagebionetworks.jira.com/wiki/spaces/IBC/pages/2741797027/GitHub+Git#Merging-a-Pull-Request

Author: There is something mentioned here: https://github.com/Sage-Bionetworks/recover/blob/main/CONTRIBUTING.md#code-review about squash and merge. i'll remove the part about rebasing since we as a team talked about picking squash and merge

> Once the pull request has been approved, we expect _you_ to merge. Although this pulls your commits into the `main` branch, it does not yet deploy your changes to the `main` production pipeline. RECOVER data has FISMA restrictions, but only our production account is FISMA compliant. Since there is no guarantee that the test data provided to us (which doesn't have FISMA restrictions) perfectly models the production dataset, we maintain a `staging` namespace in the production account which enables us to test changes on production data before pulling those changes into the `main` namespace. Merging into `main` will deploy the changes to the `staging` namespace. To complete deployment to the `main` namespace of production, we push a new tag with a specific format to this repository, which will trigger [this GitHub action](.github/workflows/upload-and-deploy-to-prod-main.yaml). There is a diagram of this process [here](https://sagebionetworks.jira.com/wiki/spaces/IBC/pages/2863202319/ETL-390+RECOVER+Integration+Testing#Implementation). See [release process](#release-process) for more details on this.

Reviewer: The content here is pretty dense - Consider bullet points for the required things that need to occur and a paragraph for the additional explanation.

### Release process

This package uses [semantic versioning](https://semver.org/) for releasing new versions.

**Background Pages:**

- See [About releases - GitHub Docs](https://docs.github.com/en/repositories/releasing-projects-on-github/about-releases) for more information about Github releases and their relationship to github tags
- See [Git - Tagging](https://git-scm.com/book/en/v2/Git-Basics-Tagging) for more information about Github tags

**Instructions:**

> 1. Once there are important features that need to be added so it’s applied to the production data, we will need to create a new tag and release

Reviewer: ```suggestion
1. Once there are important features that need to be added we will need to create a new tag and release. Once released it will be applied to production data.
```

> 1. [Create a new tag and release on the repo](https://github.com/Sage-Bionetworks/recover/releases) through here on the `main` branch. This will auto-create the new tag that you specify and release notes. Also include any known bugs for each release here.

Reviewer: ```suggestion
1. [Draft a new release for the repo here](https://github.com/Sage-Bionetworks/recover/releases), specify the `main` branch as the target, and choose a tag name. Click `Generate release notes`, review the content and be sure to include any known bugs for the release.
```

Reviewer 2: Note: may want to add that you can specify a tag name that doesn't exist to leverage the github auto tag create feature.

> 1. Wait for the CI/CD (specifically the [upload-and-deploy-to-prod-main](https://github.com/Sage-Bionetworks/recover/blob/main/.github/workflows/upload-and-deploy-to-prod-main.yaml) GH action which deploys the features to main) to finish successfully

Reviewer: ```suggestion
1. Wait for the CI/CD (specifically the [upload-and-deploy-to-prod-main](https://github.com/Sage-Bionetworks/recover/blob/main/.github/workflows/upload-and-deploy-to-prod-main.yaml) GH action which deploys the features to main) to finish successfully.
```
